svn/common.py

```diff
@@ -450,12 +453,18 @@ def diff(self, old, new, rel_path=None):
             do_combine=True)
         file_to_diff = {}
         for non_empty_diff in filter(None, diff_result.decode('utf8').split('Index: ')):
-            split_diff = non_empty_diff.split('==')
+            split_diff = \
+                    non_empty_diff.split('===================================================================')
             file_to_diff[split_diff[0].strip().strip('/')] = split_diff[-1].strip('=').strip()
         diff_summaries = self.diff_summary(old, new, rel_path)
         for diff_summary in diff_summaries:
-            diff_summary['diff'] = \
-                file_to_diff[diff_summary['path'].split(full_url_or_path)[-1].strip('/')]
+            diff_summary['diff'] = ''
```

Comment:
This section could use some commentary. Also, can you also refactor the `split()` to use startswith() or searching instead? I realize you weren't responsible for this, but the way it is is far from intuitive. Though I can surmise what we're doing here, it's not obvious when the if's will or will not trigger.